packages/monitor-v2/src/price-publisher/PublishPrices.ts

```diff
@@ -0,0 +1,158 @@
+import "@nomiclabs/hardhat-ethers";
+import { paginatedEventQuery } from "@uma/common";
+import { OracleHubEthers, OracleRootTunnelEthers, VotingV2Ethers } from "@uma/contracts-node";
+import {
+  ArbitrumParentMessenger,
+  OptimismParentMessenger,
+} from "@uma/contracts-node/dist/packages/contracts-node/typechain/core/ethers";
+import { RequestResolvedEvent } from "@uma/contracts-node/dist/packages/contracts-node/typechain/core/ethers/VotingV2";
+import { BigNumber, utils } from "ethers";
+import { logPricePublished } from "./BotLogger";
+import {
+  ARBITRUM_CHAIN_ID,
+  BLOCKS_WEEK_MAINNET,
+  Logger,
+  MonitoringParams,
+  OPTIMISM_CHAIN_ID,
+  POLYGON_CHAIN_ID,
+  getContractInstanceWithProvider,
+} from "./common";
+
+const shouldPublish = async (
+  params: MonitoringParams,
+  oracle: OracleHubEthers | OracleRootTunnelEthers,
+  event: RequestResolvedEvent
+) => {
+  const { identifier, time, ancillaryData } = event.args;
+
+  const requestHash = utils.keccak256(
+    utils.defaultAbiCoder.encode(["bytes32", "uint256", "bytes"], [identifier, time, ancillaryData])
+  );
+
+  const messagesSent = await oracle.queryFilter(oracle.filters.PushedPrice(null, null, null, null, requestHash));
```

Comment:
Not using the pagination here as we are filtering with an indexed param

Reply:
I think this makes sense for queries that are run only on mainnet. +1. Infura's requirements are a little weird (10 second timeout), so we'll have to change this if we start getting timeouts there, but I doubt we will.